--- origsrc/grafx2/src/brush_ops.c	2011-01-25 15:49:44.000000000 -0600
+++ src/grafx2/src/brush_ops.c	2011-07-03 22:31:21.426561500 -0500
@@ -24,6 +24,7 @@
 /// Code for operations about the brush (grabbing, rotating, ...) and magnifier
 //////////////////////////////////////////////////////////////////////////////
 
+#undef __STRICT_ANSI__
 #include <math.h>
 #include <stdlib.h>
 
--- origsrc/grafx2/src/graph.c	2011-03-12 17:11:04.000000000 -0600
+++ src/grafx2/src/graph.c	2011-07-03 22:31:21.464066300 -0500
@@ -25,6 +25,7 @@
 
 */
 
+#undef __STRICT_ANSI__
 #include <string.h>
 #include <stdlib.h>
 #include <math.h>
--- origsrc/grafx2/src/help.c	2011-04-04 17:19:16.000000000 -0500
+++ src/grafx2/src/help.c	2011-07-03 22:31:21.499570800 -0500
@@ -30,7 +30,7 @@
 #elif defined(__macosx__) || defined(__FreeBSD__)
     #include <sys/param.h>
     #include <sys/mount.h>
-#elif defined (__linux__)
+#elif defined (__linux__) || defined(__CYGWIN__)
     #include <sys/vfs.h>
 #elif defined(__HAIKU__)
 	#include "haiku.h"
@@ -790,7 +790,7 @@ void Button_Stats(void)
       GetDiskFreeSpaceEx(Main_current_directory,&tailleU,NULL,NULL);
       mem_size = tailleU.QuadPart;
     }
-#elif defined(__linux__) || defined(__macosx__) || defined(__FreeBSD__)
+#elif defined(__linux__) || defined(__macosx__) || defined(__FreeBSD__) || defined(__CYGWIN__)
     // Note: under MacOSX, both macros are defined anyway.
     {
       struct statfs disk_info;
--- origsrc/grafx2/src/misc.c	2011-02-06 11:16:48.000000000 -0600
+++ src/grafx2/src/misc.c	2011-07-03 22:31:21.570579800 -0500
@@ -787,7 +787,7 @@ unsigned long Memory_free(void)
   len = sizeof(maxmem);
   sysctl(mib,2,&maxmem,&len,NULL,0);
   return maxmem;
-#elif defined(__BEOS__) || defined(__HAIKU__) || defined(__SKYOS__) || defined(__amigaos4__) || defined(__TRU64__)
+#elif defined(__BEOS__) || defined(__HAIKU__) || defined(__SKYOS__) || defined(__amigaos4__) || defined(__TRU64__) || defined(__CYGWIN__)
   // No <sys/sysctl.h> on BeOS or Haiku
   // AvailMem is misleading on os4 (os4 caches stuff in memory that you can still allocate)
 #warning "There is missing code there for your platform ! please check and correct :)"
--- origsrc/grafx2/src/setup.c	2011-02-06 10:58:14.000000000 -0600
+++ src/grafx2/src/setup.c	2011-07-03 22:29:59.893708200 -0500
@@ -39,7 +39,7 @@
 #elif defined(__MINT__)
     #include <mint/osbind.h>
     #include <mint/sysbind.h>
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
   #include <limits.h>
   #include <unistd.h>
 #endif
@@ -99,7 +99,7 @@ void Set_program_directory(ARG_UNUSED co
   // Append trailing slash
   strcat(program_dir,PATH_SEPARATOR);
   // Linux: argv[0] unreliable
-  #elif defined(__linux__)
+  #elif defined(__linux__) || defined(__CYGWIN__)
   if (argv0[0]!='/')
   {
     char path[PATH_MAX];
--- origsrc/grafx2/src/text.c	2011-03-11 11:39:52.000000000 -0600
+++ src/grafx2/src/text.c	2011-07-03 22:30:39.148692900 -0500
@@ -39,7 +39,7 @@
 
 #if defined(__CAANOO__) || defined(__WIZ__) || defined(__GP2X__)
 // No X11
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__CYGWIN__)
   #include <X11/Xlib.h>
 #endif
 #endif
@@ -328,7 +328,7 @@ void Init_text(void)
 
   #elif defined(__CAANOO__) || defined(__WIZ__) || defined(__GP2X__)
   // No X11 : Only use fonts from Grafx2
-  #elif defined(__linux__)
+  #elif defined(__linux__) || defined(__CYGWIN__)
     #ifndef NOTTF
        #define USE_XLIB
     
